Flint/src/main/java/flint/CLIAdapter.java

```diff
@@ -1,65 +1,193 @@
+    } catch (FileNotFoundException e) {
+      System.out.println(flags.get("-config-jar") + " is not a valid flintConfiguration jar");
+    } catch (IllegalArgumentException e) {
+      System.out.println(flags.get("-config-class") + " is not subclass of FlintConfiguration");
+    } catch (ClassNotFoundException e) {
+      System.out.println(flags.get("-config-class") + " is not found in " + flags.get("-config-jar"));
+    } catch (Exception e) {
+      System.out.println("Died from the unexpected");
+    } finally {
+      System.exit(0);
+    }
+
+    // check if the file to be lint is a java file
+    if (!isJava(flags.get("-file-path"))) {
+      System.out.println(flags.get("-file-path") + " is not a valid Java file.");
+
+      System.exit(0);
     }
 
-    /**
-     * This function initializes and returns the FlintConfiguration class.
-     *
-     * @param configPath - The path that leads to the FlintConfiguration subclass file.
-     * @return Returns the initialized FlintConfiguration subclass if passed in a valid path.
-     * @throws FileNotFoundException - If configPath did not lead to a file.
-     * @throws IllegalArgumentException - If the passed in configPath is not a FlintConfiguration subclass.
-     * */
-    private static FlintConfiguration configInit(String configPath) throws Exception {
-        throw new Exception("Not Yet Implemented");
+    // if all input matches requirement, start linting
+    try {
+      result = run(flags.get("-file-path"), config);
+    } catch (Exception e) {
+      // not sure what kind of exceptions will happen
+    } finally {
+      System.exit(0);
     }
 
-    /**
-     * This function checks if filePath leads to an existing java file.
-     *
-     * @param filePath - The file path to check.
-     * @return Returns the true if the file exists and is a java file. Otherwise false.
-     * */
-    private static boolean isJava(String filePath) throws Exception {
-        throw new Exception("Not Yet Implemented");
+    resultOutput(result);
+  }
+
+  /**
+  * This function initializes and returns the FlintConfiguration class.
+  *
+  * @param configPath - The path that leads to the FlintConfiguration subclass jar.
+  * @param className - The java style class name for the FlintConfiguration subclass.
+  * @return Returns the initialized FlintConfiguration subclass if passed in a valid path.
+  * @throws FileNotFoundException - If configPath did not lead to a jar.
+  * @throws IllegalArgumentException - If the passed in configPath is not a FlintConfiguration subclass.
+  * @throws ClassNotFoundException - If the className could not be found
+  * */
+  public static FlintConfiguration configInit(String configPath, String className) throws Exception {
+    File file = new File(configPath);
+
+    if (!file.exists()) {
+      throw new FileNotFoundException();
+    }
+
+    String realConfigPath = configPath;
+
+    if (realConfigPath.charAt(0) != '\\' && realConfigPath.charAt(0) != '/') {

```

Comment:
I have tested that, it it just not something we can put it on the test case since we don't have platform based test env for me to put those specific test. I think I have some of them in some of my commits.